### PR TITLE
Implements Swift 4 support

### PR DESCRIPTION
**Pull request by annjose**
This PR fixes issue https://github.com/delba/TextAttributes/issues/27 by implementing Swift 4.2 support to the project. It has migrated all the source code and all project targets (iOS, MacOS, watch and tvOS) to Swift 4.2.
This PR requires Xcode 10 with Swift 4.2 to build successfully.

There is no change in the API surface and all tests are passing.
Please review and merge.
 
@delba : If you could add the label `hacktoberfest` to this PR and the above issue, this would qualify for the [Hacktoberfest contest](https://hacktoberfest.digitalocean.com/) and give some credits.

### CODE REVIEW

File: Tests/TextAttributesTests.swift [hunk 1]
```diff
@@ -33,7 +33,7 @@ class TextAttributesTests: XCTestCase {
         let color = Color(white: 0.42, alpha: 1)
         let ligature: LigatureStyle = .all
         let float: CGFloat = 0.42
-        let underlineStyle: NSUnderlineStyle = .styleSingle
+        let underlineStyle: NSUnderlineStyle = NSUnderlineStyle.single
```
Comment (cruisediary) [inline Tests/TextAttributesTests.swift]: How about using `.single` ?

**Comment by cruisediary** [comment]
Update needed
- .swift-version 3.0 to 4.2
- .travis.yml using Xcode 10 and iphonesimulator12

File: Tests/TextAttributesTests.swift (hunk 1 above)
Comment (annjose) [reply Tests/TextAttributesTests.swift]: Good catch. Will change it.

**Comment by annjose** [comment]
@cruisediary - Updated swift version and travis config.

**Comment by cruisediary** [comment]
@delba please review this PR 😄

**Comment by max-potapov** [comment]
@delba could you please review this changes?

**Comment by weiran** [comment]
Thank you @annjose for updating this project to Swift 4.2. As @delba hasn't been actively maintaining this repo for a while, I'm now using your fork:

```
pod 'TextAttributes', :git => 'https://github.com/annjose/TextAttributes', :branch => 'annjose-master'
```

**Comment by afturner** [comment]
@delba would you be interested in a maintainer?

**Comment by howardck** [comment]
If you’re asking if I’d be interested in being a maintainer, the answer is unfortunately no — far too busy!
Best,
Howard Katz

**Comment by delba** [comment]
Hi @annjose, thank you very for the PR, the lib has been updated for the latest Swift version.
Thanks again :)

**Pull request closed without merging**